File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/IO/NonFlushingStorage.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.IO;
using Sharpen.Lang;

namespace Db4objects.Db4o.IO
{
	/// <summary>
	/// Storage adapter that does not pass flush calls
	/// on to its delegate.
	/// </summary>
	/// <remarks>
	/// Storage adapter that does not pass flush calls
	/// on to its delegate.
	/// You can use this
	/// <see cref="IStorage">IStorage</see>
	/// for improved db4o
	/// speed at the risk of corrupted database files in
	/// case of system failure.
	/// </remarks>
	public class NonFlushingStorage : StorageDecorator
	{
		public NonFlushingStorage(IStorage storage) : base(storage)
		{
		}

		protected override IBin Decorate(BinConfiguration config, IBin storage)
		{
			return new NonFlushingStorage.NonFlushingBin(storage);
		}

		private class NonFlushingBin : BinDecorator
		{
			public NonFlushingBin(IBin storage) : base(storage)
			{
			}

			public override void Sync()
			{
			}

			public override void Sync(IRunnable runnable)
			{
				runnable.Run();
			}
		}
	}
}

File: db4oj/core/src/com/db4o/io/NonFlushingStorage.java
/* Copyright (C) 2008  Versant Inc.  http://www.db4o.com */
package com.db4o.io;

/**
 * Storage adapter that does not pass flush calls 
 * on to its delegate.
 * You can use this {@link Storage} for improved db4o
 * speed at the risk of corrupted database files in 
 * case of system failure.    
 */
public class NonFlushingStorage extends StorageDecorator {

	public NonFlushingStorage(Storage storage) {
		super(storage);
    }

	@Override
	protected Bin decorate(BinConfiguration config, Bin storage) {
		return new NonFlushingBin(storage);
	}
	
	private static class NonFlushingBin extends BinDecorator {

		public NonFlushingBin(Bin storage) {
			super(storage);
	    }
		
		@Override
		public void sync() {
		}
		
		@Override
		public void sync(Runnable runnable) {
			runnable.run();
		}
		
		
	}

}
